### ISSUE TITLE
Use torch.gather instead of to_one_hot for selecting by indices

### ISSUE THREAD

**Issue by Pastafarianist** [yandexdataschool/Practical_RL#182]
Example for `week6/a2c-optional`:

        actions = torch.tensor(trajectory['actions'], device=device)
        action_log_probs = torch.gather(
            trajectory['log_probs'],
            dim=-1,
            index=actions.unsqueeze(-1)).squeeze(-1)

This _should_ work more efficiently than `to_one_hot`, although to be certain we could also benchmark it.
